stubs/SQLAlchemy/sqlalchemy/util/compat.pyi

```diff
@@ -24,21 +25,32 @@ from urllib.parse import (
 
 byte_buffer = BytesIO
 
+py312: bool
+py311: bool
+py310: bool
 py39: bool
 py38: bool
 py37: bool
-py3k: Literal[True]
-py2k: Literal[False]
+py3k: Final = True
+py2k: Final = False
 pypy: bool
 cpython: bool
-win32: bool
-osx: bool
+if sys.platform == "win32":
+    win32: Final = True
+else:
+    win32: Final = False
+if sys.platform == "darwin":
+    osx: Final = True
+else:
+    osx: Final = False
 arm: bool
+is64bit: bool
 has_refcount_gc: bool
-contextmanager = contextlib.contextmanager
 dottedgetter = operator.attrgetter
-namedtuple = collections.namedtuple  # noqa: Y024
-next = builtins.next
+
+class StringIO(_StringIO):
+    def seek(self, __pos: int, __whence: int = 0) -> int: ...
+    def truncate(self, __pos: int | None = None) -> int: ...
```

Comment:
I don't see these methods being overridden at runtime on the `rel_1_4` branch?

Reply:
Stubtest probably emits errors here for the same reason as it emits errors on the stdlib:

https://github.com/python/typeshed/blob/7f986bdf859a5c009474356addb7420b0a79844b/tests/stubtest_allowlists/py3_common.txt#L111-L112

I'm planning on working on reducing stubtest errors for re-exported objects after mypy 1.0 is released. (There are enough stubtest changes on mypy `master` already; working on it now would make upgrading too disruptive for us imo.)

Reply:
As far as I can tell it should be using StringIO directly. But I kindof assumed sqlalchemy did some magic I hadn't spotted. And I think I manually checked at runtime? (not sure if I did for this one).

Makes sense if the same issue happens in IO. I'll allowlist them and this can be fixed with an eventual mypy/stubtest update.